### PR TITLE
Fixed #25274 --- Made inspectdb handle renamed fields in unique_together.

### PR DESCRIPTION
**Pull request by jbzdak**
Inspectdb now handles renamed fields in unique_together. 

It was tested on sqlite and postgresql (I don't know whether I'm supposed to check it on other all databases).

### CODE REVIEW

**Comment by charettes** [comment]
Buildbot, add to whitelist.

File: tests/inspectdb/models.py [hunk 1]
```diff
@@ -73,5 +73,14 @@ class UniqueTogether(models.Model):
...
+
+    non_unique = models.IntegerField(db_column="non__unique_column")
+    non_unique_0 = models.IntegerField(db_column="non_unique__column")
+
     class Meta:
-        unique_together = ('field1', 'field2')
+        unique_together = [
+            ('field1', 'field2'),
+            ('from_field', 'field1'),
+            ('non_unique', 'non_unique_0')
```
Comment (charettes) [inline tests/inspectdb/models.py]: Add a trailing comma.

File: django/core/management/commands/inspectdb.py [hunk 2]
```diff
@@ -71,6 +71,7 @@ def handle_inspection(self, options):
                 except NotImplementedError:
                     constraints = {}
                 used_column_names = []  # Holds column names used in the table so far
+                column_to_field_name = {}  # Maps original column names to names of django fields
```
Comment (timgraham) [inline django/core/management/commands/inspectdb.py]: django -> model

File: django/core/management/commands/inspectdb.py [hunk 3]
```diff
@@ -255,7 +257,8 @@ def get_meta(self, table_name, constraints):
                 if len(columns) > 1:
                     # we do not want to include the u"" or u'' prefix
                     # so we build the string rather than interpolate the tuple
-                    tup = '(' + ', '.join("'%s'" % c for c in columns) + ')'
+                    tup = '(' + ', '.join(
+                        "'%s'" % column_to_field_name[c] for c in columns) + ')'
```
Comment (timgraham) [inline django/core/management/commands/inspectdb.py]: keeping this on a single line is okay (we allow up to 119 characters when it helps readability)

File: tests/inspectdb/models.py [hunk 4]
```diff
@@ -73,5 +73,14 @@ class UniqueTogether(models.Model):
     field1 = models.IntegerField()
     field2 = models.CharField(max_length=10)
```
Comment (timgraham) [inline tests/inspectdb/models.py]: I think the blank lines could be omitted between the model fields.

File: tests/inspectdb/tests.py [hunk 5]
```diff
@@ -260,3 +249,57 @@ def test_custom_fields(self):
             self.assertIn("big_int_field = models.BigIntegerField()", output)
         finally:
             connection.introspection.data_types_reverse = orig_data_types_reverse
+
+
+class TestUniqueTogether(TestCase):
+
```
Comment (timgraham) [inline tests/inspectdb/tests.py]: no blank line needed before/after docstring or at the start of each test

File: tests/inspectdb/tests.py [hunk 6]
```diff
@@ -260,3 +249,57 @@ def test_custom_fields(self):
...
+    from columns whose names were cleaned.
+    """
+
+    def setUp(self):
+        out = StringIO()
+        call_command('inspectdb',
+                     table_name_filter=lambda tn: tn.startswith('inspectdb_uniquetogether'),
+                     stdout=out)
+        self.inspect_output = out.getvalue()
+        # self.unique_re = re.compile(r".*unique_together = \(([\'\w\s,\d_\(\)]+),\).*")
```
Comment (timgraham) [inline tests/inspectdb/tests.py]: chop?

File: tests/inspectdb/tests.py [hunk 7]
```diff
@@ -260,3 +249,57 @@ def test_custom_fields(self):
...
+        self.unique_re = re.compile(r".*unique_together = \((.+),\).*")
+
+    def test_unique_together_meta(self):
+
+        self.assertIn("unique_together", self.inspect_output,
+            msg='inspectdb should generate unique_together.')
```
Comment (timgraham) [inline tests/inspectdb/tests.py]: I think I would put the "msg" as a comment before each assertion.

**Comment by jbzdak** [comment]
I have added suggested changes, two questions: 

1. Should I squash those two commits
2. After browsing code for second time I'm wondering if tests could be done better. Here is the issue: Tests implicitly test for the order of unique_togetgher entries so if two columns were present in unique_together in reverse order ("foo", "bar") instead of ("bar", "foo") this would be unittest error, while at the database level these two constraints are equivalent (as far as I know). I could amend tests to fix this behaviour, but it sould decrease test clarity. Is this worthwhile?

**Comment by timgraham** [comment]
Splitting up the test into a class didn't seem to add clarity, so I moved it back into a method. I didn't look closely about the issue with field name ordering, but I haven't seen the test failure non-deterministically so I think we're okay for now. Merged in 2cb50f935aa70e91dd6c2f253becd636a2eb6fb7, thanks!

**Pull request closed without merging**